build-caching/src/main/java/com/vertispan/j2cl/build/DiskCache.java

```diff
@@ -170,6 +170,10 @@ private void checkForWork() {
         } catch (IOException e) {
             // disaster, can't interact with the cache, stop and give up
             // TODO mark all pending work as canceled?
+        } catch (ClosedWatchServiceException e) {
+            // This is purely noise in the log and doesn't indicate an actual
```

Comment:
```suggestion
            if(!livenessThread.getState().equals(Thread.State.TERMINATED)) {
                 throw new Error(e);
            }

            // This is purely noise in the log and doesn't indicate an actual
```

Reply:
I think that we can add an additional check to ensure that this exception is thrown only when the build is complete. This is necessary to rule out other scenarios.